pycalphad/core/equilibrium.py

```diff
@@ -101,7 +103,7 @@ def _eqcalculate(dbf, comps, phases, conditions, output, data=None, per_phase=Fa
     # For each phase select all conditions where that phase exists
     # Perform the appropriate calculation and then write the result back
     for phase in active_phases:
-        dof = sum([len(x) for x in dbf.phases[phase].constituents])
+        dof = len(models[phase].site_fractions)
```

Comment:
This makes `models` a required argument, right?

Reply:
Yes.